src/psyclone/psyir/nodes/call.py

```diff
@@ -593,3 +610,188 @@ def _location_txt(node):
             f"Failed to find a Routine named '{rsym.name}' in "
             f"{_location_txt(root_node)}. This is normally because the routine"
             f" is within a CodeBlock.")
+
+    def _check_argument_type_matches(
+                self,
+                call_arg: DataSymbol,
+                routine_arg: DataSymbol,
+            ) -> bool:
+        """Return information whether argument types are matching.
+        This also supports 'optional' arguments by using
+        partial types.
+
+        :param call_arg: _description_
+        :type call_arg: DataSymbol
+        :param routine_arg: _description_
+        :type routine_arg: DataSymbol
+        :raises CallMatchingArgumentsNotFound: _description_
```

Comment:
Blank line before :raises: please and please populate the description. You could just have one :raises: as they both essentially mean 'no match'.

Reply:
OK.